src/team/stats.ts

```diff
@@ -0,0 +1,185 @@
+  const winPts: number = stats.wins / 30;
+  if (stats.fkdr > 10) {
+    fkdrPts = 0.1 * Math.pow(stats.fkdr, 2) + 2.5 * stats.fkdr + 35;
+  }
+  if (stats.wlr > 10) {
+    wlrPts = 45 * stats.wlr - 134;
+  }
+  if (stats.bblr > 10) {
+    bblrPts = 45 * stats.bblr - 134;
+  }
+  const score =
+    fkdrPts + starPts + wlrPts + bblrPts + finalPts + bedPts + winPts;
+  return score * 10;
+}
+
+export default async function stats(
+  msg: Message,
+  args: TeamArgs
+): Promise<void> {
+  let teams;
+  await msg.react("👍");
+  if (args.splitCommand[2]) {
+    const specifiedTag = args.splitCommandLower[2].toUpperCase();
+    teams = await listTeams({ tag: specifiedTag });
+    if (!teams.length) {
+      await msg.channel.send(
+        errorEmbed()
+          .setTitle("Invalid team tag")
+          .setDescription(`A team with tag \`${specifiedTag}\` doesn't exist.`)
+      );
+      return;
+    }
+  } else {
+    const userMembership = await listTeamMemberships({
+      user_id: msg.author.id,
+    });
+
+    if (!userMembership.length) {
+      await msg.channel.send(
+        errorEmbed()
+          .setTitle("No team found")
+          .setDescription(
+            "In order to use this command without a team tag, you need to be in a team.\nTry `!team stats TAG`."
+          )
+      );
+      return;
+    }
+    teams = await listTeams({ id: userMembership[0].team_id });
+  }
+  const members = await listTeamMemberships({ team_id: teams[0].id });
+  const memberIds = members.map((item) => item.user_id);
+  const uuids = [];
+  for (let i = 0; i < memberIds.length; i++) {
+    const uuid = await listUsers({ discord_id: memberIds[i] });
+    uuids.push(uuid[0].minecraft_uuid);
+  }
+  const stats = [];
+  for (let i = 0; i < uuids.length; i++) {
+    const player = await fetchBedwarsData(uuids[i]);
+    stats.push(player);
+  }
+  const teamData = {
+    finals: 0,
+    beds: 0,
+    wins: 0,
+    games: 0,
+    fDeaths: 0,
+    bLost: 0,
+    losses: 0,
+    fkdr: 0,
+    bblr: 0,
+    wlr: 0,
+    winPercent: 0,
+    rating: 0,
+    stars: 0,
+  };
+  for (let i = 0; i < stats.length; i++) {
+    teamData.finals += stats[i].finals;
+    teamData.beds += stats[i].beds;
+    teamData.wins += stats[i].wins;
+    teamData.games += stats[i].games;
+    teamData.fDeaths += stats[i].fDeaths;
+    teamData.bLost += stats[i].bLost;
+    teamData.losses += stats[i].losses;
+    teamData.fkdr += stats[i].fkdr;
+    teamData.bblr += stats[i].bblr;
+    teamData.wlr += stats[i].wlr;
+    teamData.stars += stats[i].stars;
+  }
+  teamData.fkdr /= stats.length;
+  teamData.bblr /= stats.length;
+  teamData.wlr /= stats.length;
+  teamData.winPercent = (teamData.wlr * 100) / (teamData.wlr + 1);
+  teamData.rating = calculateRating(teamData);
+  Object.keys(teamData).map(function (key) {
+    if (!(teamData[key] === Math.round(teamData[key]))) {
+      teamData[key] = parseFloat(teamData[key].toFixed(2));
+    }
+  });
```

Comment:
```suggestion
  const teamData = {
    finals: 0,
    beds: 0,
    wins: 0,
    games: 0,
    fDeaths: 0,
    bLost: 0,
    losses: 0,
    fkdr: 0,
    bblr: 0,
    wlr: 0,
    winPercent: 0,
    rating: 0,
    stars: 0,
  };
  
  const teamStats = stats.reduce((acc, cur) => { 
      acc.finals += stats[i].finals;
      acc.beds += stats[i].beds;
      acc.wins += stats[i].wins;
      acc.games += stats[i].games;
      acc.fDeaths += stats[i].fDeaths;
      acc.bLost += stats[i].bLost;
      acc.losses += stats[i].losses;
      acc.fkdr += stats[i].fkdr;
      acc.bblr += stats[i].bblr;
      acc.wlr += stats[i].wlr;
      acc.stars += stats[i].stars;
      
      return acc;
   }, teamData);
  
  teamData.fkdr /= stats.length;
  teamData.bblr /= stats.length;
  teamData.wlr /= stats.length;
  teamData.winPercent = (teamData.wlr * 100) / (teamData.wlr + 1);
  teamData.rating = calculateRating(teamData);
  Object.keys(teamData).map(function (key) {
    if (!(teamData[key] === Math.round(teamData[key]))) {
      teamData[key] = parseFloat(teamData[key].toFixed(2));
    }
  });
```


yeah im just gonna call you about this

Reply:
done